File: Data Structures & Algorithms/max-consecutive-ones/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums) {
        let max = 0;
        let cur = 0
        for (let i = 0; i < nums.length; i++) {
            if (nums[i] === 1) {
                cur++
                if (cur > max) {
                    max = cur
                }
            } else {
                cur = 0
            }
        }
        return max
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low = 1
        high = max(piles)
        cur = high

        while low <= high:
            mid = (low + high) // 2
            if self.speedTest(piles, h, mid):
                cur = mid
                high = mid - 1
            else:
                low = mid + 1
        
        return cur


    def speedTest(self, piles, h, k):
        for pile in piles:
            h -= math.ceil(pile / k)
        if h >= 0:
            return True
        return False

